mongoquery/__init__.py

```diff
@@ -29,7 +29,10 @@ def _match(self, condition, entry):
                 for sub_operator, sub_condition in condition.items()
             )
         else:
-            if isinstance(entry, Sequence):
+            # strings are a subtype of Sequence, thus checking first
+            if isinstance(entry, string_type):
+                return condition == entry
+            elif isinstance(entry, Sequence):
```

Comment:
Looks like this would be the 3rd place where this condition is required.

Maybe we should have an is_non_string_sequence() function (better name ideas very welcome) and use that?